### PR TITLE
Modify container in GetInvokeResult method

### PR DESCRIPTION
**Pull request by chenzhitong**
close https://github.com/neo-project/neo-modules/issues/389

### CODE REVIEW

**Comment by cloud8little** [comment]
Test failed. 
```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "script": "DBTZkoZEWksPf18rLLJIQntret3w6hHADBJnZXRldGhwcmljZVJlcXVlc3QMFFxqUwrI8ncxljz9i8O7wfDg8Y6+QWJ9W1I=",
    "state": "FAULT",
    "gasconsumed": "1039070",
    "exception": "Value cannot be null. (Parameter 'source')",
    "stack": []
  }
}
```

bef63c9ee71c89f73a8f3833448e1fb
75ab24a5ff238d46ab0eff4677b64ef

**Comment by chenzhitong** [comment]
I tested it successfully.

image

**Comment by cloud8little** [comment]
Example Request:
```
{
  "jsonrpc": "2.0",
  "method": "invokefunction",
  "params": [ "0xdbe814bc1536cd5b7507c3ce5ed690a057ea83b8", "getethpriceRequest",
 [
 {"type": "Hash160", "value": "0xeaf0dd7a6b7b4248b22c2b5f7f0f4b5a448692d9"}
],
	[
      {
        "account": "NfkPM5juAxBETYXNMZpp1zzxVCwJpAqQRh",
        "scopes": "CustomContracts",
			"allowedcontracts":["0xdbe814bc1536cd5b7507c3ce5ed690a057ea83b8","0xde5f57d430d3dece511cf975a8d37848cb9e0525"]
      }
    ]
],
  "id": 1
}
```

Contract:
```
      public static void GetethpriceRequest(byte[] from)
        {
            if (!Runtime.CheckWitness(from)) return;
            string url = "http://192.168.0.106:5000/api/get/ethprice";
            string filter = "$.result";
            string callback = "callback_eth";
            object userdata = from; // arbitrary type

            if (!NEO.Transfer(from, to, 5))
            {
                Runtime.Log("transfer failed");
            }

            long gasForResponse = Oracle.MinimumResponseFee * 10;

            Oracle.Request(url, filter, callback, userdata, gasForResponse);
        }
```
the problem is Runtime.CheckWitness failed. please have a check.

**Comment by chenzhitong** [comment]
Oh, you're right. 
Also what is the format of the oracle response, do you have an example of a server response?

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void SerializeUnsigned(BinaryWriter writer)
 
         private JObject GetInvokeResult(byte[] script, Signers signers = null)
         {
-            using ApplicationEngine engine = ApplicationEngine.Run(script, container: signers, gas: settings.MaxGasInvoke);
+            using ApplicationEngine engine = ApplicationEngine.Run(script, container: new Transaction() { Witnesses = signers.Witnesses }, gas: settings.MaxGasInvoke);
```
Comment (cloud8little) [inline src/RpcServer/RpcServer.SmartContract.cs:66]:
```suggestion
            Transaction tx = new Transaction
            {
                Signers = signers.GetSigners(),
                Attributes = Array.Empty<TransactionAttribute>(),
                Witnesses = signers.Witnesses,
            };
            using ApplicationEngine engine = ApplicationEngine.Run(script, container: tx, gas: settings.MaxGasInvoke);
```

**Review by Tommo-L — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed:

1. Invoke nep5 transfer.
b5eb0c75c014b5423492eec790d0a29

2. Invoke Oracle contract
a1f84aa0adc5ee19064fc85be3b6830

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
Merge?

**Comment by chenzhitong** [comment]
image

**Pull request merged**